src/link-manager.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @fileoverview Link manager for controlling the link query.
+ */
+
+/** @const */
+genotet.linkManager = {};
+
+/**
+ * Object that stores the link views by their type.
+ */
+/** @type {!Object<!Array<{
+ *    target: genotet.View,
+ *    action: string,
+ *    response: string
+ *  }>>}
+ */
+genotet.linkManager.links = {};
+
+/** @const {!Array<string>} */
+genotet.linkManager.EXPRESSION_ACTIONS = ['addGeneProfile'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.NETWORK_ACTIONS = ['nodeClick', 'edgeClick'];
+
+/** @const {!Array<string>} */
+genotet.linkManager.BINDING_ACTIONS = ['updateTrack', 'locus'];
+
+/** @const {string} */
+genotet.linkManager.DEFAULT_MAPPING_FILENAME = 'namecode';
+
+/**
+ * Initializes the link manager.
+ */
+genotet.linkManager.init = function() {
+  genotet.linkManager.links = {};
+  genotet.data.geneBindingMappingFile =
+    genotet.linkManager.DEFAULT_MAPPING_FILENAME;
+};
+
+
+/**
+ * Set up the link manager.
+ */
+genotet.linkManager.link = function() {
+  for (var linkViewName in genotet.linkManager.links) {
+    var linkView = genotet.viewManager.views[linkViewName];
+    $(linkView)
+      .on('genotet.link', function(event, linkData) {
+        switch (linkData.action) {
+          case 'nodeClick':
+            var node = linkData.data;
+            var genes = [node.id];
+            break;
+          case 'edgeClick':
+            var edge = linkData.data;
+            var genes = edge.id.split(',');
+            break;
+        }
+        genotet.linkManager.links[linkViewName]
+          .filter(function(object) {
+            return object.action == linkData.action;
+          })
+          .forEach(function(object) {
+            var targetView = object.target;
+            switch (object.response) {
+              case 'addGeneProfile':
+                genes.forEach(function(gene) {
+                  targetView.signal('link', {
+                    response: object.response,
+                    data: gene
+                  });
+                });
+                break;
+              case 'updateTrack':
+                var sourceGene = genes[0];
+
+                var params = {
+                  type: 'mapping',
+                  fileName: genotet.data.geneBindingMappingFile
+                };
+                $.get(genotet.data.serverURL, params, function(data) {
```

Comment:
So why do you load all the list when update panels and dialogs? So you just realized that list should only be loaded after upload?

However, the upload part is not finished yet. The 'data uploaded successfully' signal still doesn't work well now. Jiaming will upgrade this part, and I don't know when the data would be uploaded successfully until he finish this part.